src/raftstore/store/peer.rs

```diff
@@ -721,10 +721,14 @@ impl Peer {
     /// Collects all pending peers and update `peers_start_pending_time`.
     pub fn collect_pending_peers(&mut self) -> Vec<metapb::Peer> {
         let mut pending_peers = Vec::with_capacity(self.region().get_peers().len());
-        let status = self.raft_group.status();
+        let status = self.raft_group.status_ref();
```

Comment:
If it is possible the status changes before we use it?

Reply:
No, it can't. Lifetime ensure it. Magic of rust.